src/SFA.DAS.Employer.PR.Web/Controllers/YourTrainingProvidersController.cs

```diff
@@ -21,17 +21,39 @@ public async Task<IActionResult> Index([FromRoute] string employerAccountId, Can
         _sessionService.Delete<AddTrainingProvidersSessionModel>();
 
         var response = await _outerApiClient.GetAccountLegalEntities(employerAccountId, cancellationToken);
-        var accountLegalEntities = response.AccountLegalEntities;
+        var accountLegalEntities = response.AccountLegalEntities.OrderBy(a => a.Name);
 
-        var sessionModel = new AddTrainingProvidersSessionModel
+        var legalEntities = new List<AccountLegalEntity>();
+
+        foreach (var ale in accountLegalEntities)
+        {
+            var permissions = new List<Permission>();
+            permissions.AddRange(ale.Permissions.OrderBy(p => p.ProviderName));
+            ale.Permissions = permissions;
+            legalEntities.Add(ale);
+        }
+
+
+        var permissionsJustAdded = false;
+        string? providerJustAdded = null;
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+        if (sessionModel?.SuccessfulAddition != null && sessionModel.SuccessfulAddition)
         {
-            AccountLegalEntities = accountLegalEntities
+            permissionsJustAdded = true;
+            providerJustAdded = sessionModel.ProviderName;
+        }
+
+        sessionModel = new AddTrainingProvidersSessionModel
```

Comment:
The user may not event start Add TP journey here, so why add session model here? I think the journey starts on selection of legal entity, so there should not be any session model till the legal entity is selected.

Reply:
yes I've fixed this so you should be able to continue the review